### ISSUE TITLE
bug: 'no-useless-constructor recognizes the required constructor of subclass as the error.

### ISSUE THREAD

**Issue by saneyuki** [eslint/eslint#4925]
## env
- 2.0.0-beta.1

## steps to reproduce

1. Enable `no-useless-constructor` as warning/error
2. Lint this code

```javascript
// filename: test.jsx

import * as React from 'react';

export class TestComponent extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div></div>
        );
    }
}
```

## result

- the error message is:
  - `7:5 error Useless constructor no-useless-constructor`
- In this case, `TestComponent`'s constructor must be called, so it should not be the error.

**Comment by eslintbot** [comment]
@saneyuki Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
JavaScript will provide a default constructor that will call super by default. So you don't need constructor in this case. I think this is correct behavior.

**Comment by michaelficarra** [comment]
Yep, looks like the rule is working fine.
